muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceJournalBackingBean.java

```diff
@@ -130,19 +163,66 @@ public Long getWorkspaceJournalEntryId() {
   public void setWorkspaceJournalEntryId(Long workspaceJournalEntryId) {
     this.workspaceJournalEntryId = workspaceJournalEntryId;
   }
-
-  private Long workspaceEntityId;
   
+  public void setStudentId(Long studentId) {
+    this.studentId = studentId;
+  }
+  
+  public Long getStudentId() {
+    return studentId;
+  }
+  
+  public boolean isMyJournal() {
+    return studentId == null;
+  }
+  
+  public boolean isCanListAllEntries() {
+    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
+    return sessionController.hasCoursePermission(
+        MuikkuPermissions.LIST_ALL_JOURNAL_ENTRIES, workspaceEntity);
+  }
+  
+  public List<UserWithUserEntity> getWorkspaceStudents() {
+    ArrayList<UserWithUserEntity> result = new ArrayList<>();
+    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
+    List<User> userList = workspaceController.listUsersByWorkspaceEntityAndRoleArchetype(
+        workspaceEntity,
+        WorkspaceRoleArchetype.STUDENT);
+    
+    for (User user : userList) {
+      UserEntity userEntity =
+          userEntityController.findUserEntityByDataSourceAndIdentifier(
+              user.getSchoolDataSource(),
+              user.getIdentifier());
+      result.add(
+          new UserWithUserEntity(user, userEntity)
+      );
+    }
+    
+    return result;
+  }
+
   public List<WorkspaceJournalEntry> getJournalEntries() {
     WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
```

Comment:
Initialize in PostConstruct and just return it here, this solution can lead to performance issues